storm-core/src/jvm/org/apache/storm/cluster/PaceMakerStateStorage.java

```diff
@@ -0,0 +1,212 @@
+        this.pacemakerClient = pacemakerClient;
+        this.stateStorage = stateStorage;
+    }
+
+    @Override
+    public String register(ZKStateChangedCallback callback) {
+        return stateStorage.register(callback);
+    }
+
+    @Override
+    public void unregister(String id) {
+        stateStorage.unregister(id);
+    }
+
+    @Override
+    public String create_sequential(String path, byte[] data, List<ACL> acls) {
+        return stateStorage.create_sequential(path, data, acls);
+    }
+
+    @Override
+    public void mkdirs(String path, List<ACL> acls) {
+        stateStorage.mkdirs(path, acls);
+    }
+
+    @Override
+    public void delete_node(String path) {
+        stateStorage.delete_node(path);
+    }
+
+    @Override
+    public void set_ephemeral_node(String path, byte[] data, List<ACL> acls) {
+        stateStorage.set_ephemeral_node(path, data, acls);
+    }
+
+    @Override
+    public Integer get_version(String path, boolean watch) throws Exception {
+        return stateStorage.get_version(path, watch);
+    }
+
+    @Override
+    public boolean node_exists(String path, boolean watch) {
+        return stateStorage.node_exists(path, watch);
+    }
+
+    @Override
+    public List<String> get_children(String path, boolean watch) {
+        return stateStorage.get_children(path, watch);
+    }
+
+    @Override
+    public void close() {
+        stateStorage.close();
+        pacemakerClient.close();
+    }
+
+    @Override
+    public void set_data(String path, byte[] data, List<ACL> acls) {
+        stateStorage.set_data(path, data, acls);
+    }
+
+    @Override
+    public byte[] get_data(String path, boolean watch) {
+        return stateStorage.get_data(path, watch);
+    }
+
+    @Override
+    public Map get_data_with_version(String path, boolean watch) {
+        return stateStorage.get_data_with_version(path, watch);
+    }
+
+    @Override
+    public void set_worker_hb(String path, byte[] data, List<ACL> acls) {
+        int retry = maxRetries;
+        while (true) {
+            try {
+                HBPulse hbPulse = new HBPulse();
+                hbPulse.set_id(path);
+                hbPulse.set_details(data);
+                HBMessage message = new HBMessage(HBServerMessageType.SEND_PULSE, HBMessageData.pulse(hbPulse));
+                HBMessage response = pacemakerClient.send(message);
+                if (response.get_type() != HBServerMessageType.SEND_PULSE_RESPONSE) {
+                    throw new HBExecutionException("Invalid Response Type");
+                }
+                LOG.debug("Successful set_worker_hb");
+                break;
+            } catch (Exception e) {
+                if (retry <= 0) {
+                    throw Utils.wrapInRuntime(e);
+                }
+                LOG.error("{} Failed to set_worker_hb. Will make {} more attempts.", e.getMessage(), retry--);
```

Comment:
Lets move decrementing retry outside of the log message.  It is too easy to delete/comment out a log message and if there are side effects it becomes a problem.